cfg_mgmt/reporting.py

```diff
@@ -30,6 +30,60 @@ class CfgElementStatusReport:
     status: typing.Optional[cmm.CfgStatus]
 
 
+def analyse_cfg_element_status(
+    cfg_element_status: CfgElementStatusReport,
+) -> cmm.CfgStatusAnalysis:
+
+    analysis = cmm.CfgStatusAnalysis(
```

Comment:
as a general pattern, I would rather collect the state in variables, and create the result object at the end (rather than mutating its state)